app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,24 @@
+class AccountActivationsController < ApplicationController
+  before_action :get_user, only: :edit
+
+  def edit
+    if @user.authenticated?(:activation, params[:id]) && !@user.activated?
+      @user.activate
+      log_in @user
+      flash[:success] = t "users.edit.account_activated"
+      redirect_to @user
+    else
+      flash[:danger] = t "users.edit.invalid_activation_link"
+      redirect_to root_path
+    end
+  end
+
+  private
+  def get_user
+    @user = User.find_by email: params[:email]
+    if @user.nil?
+      flash[:danger] = t "users.new.failed"
+      redirect_to root_path
+    end
```

Comment:
chị có suggest lại code này chứ code ko sai
mục đích là trả về kết quả sớm nhất cho method nhé

```suggestion
    return if @user
    
    flash[:danger] = t "users.new.failed"
    redirect_to root_path
```